src/components/QueryForm.tsx

```diff
@@ -0,0 +1,248 @@
+          break;
+        case 'Assessment tool':
+          setAssessmentTool(value);
+          break;
+        case 'Imaging modality':
+          setImagingModality(value);
+          break;
+        default:
+          break;
+      }
+    }
+    
+    function updateContinuousQueryParams(fieldLabel: string, value: string | null) {
+      switch (fieldLabel) {
+        case 'Min age':
+          setMinAge(value);
+          break;
+        case 'Max age':
+          setMaxAge(value);
+          break;
+        case 'Minimum number of sessions':
+          setMinNumSessions(value);
+          break;
+        default:
+          break;
+      }
+    }
+  
+    function setQueryParam(param: string, value: FieldInput, searchParamsObject: URLSearchParams) {
+      if (Array.isArray(value)) {
+        value.forEach((v) => {
+          searchParamsObject.append(param, v.id);
+        })
+      }
+      else {
+        searchParamsObject.set(param, value?.id ?? '');
+      }
+    }
+  
+    function constructQueryURL() {
+      const queryParams = new URLSearchParams();
+  
+      setQueryParam('node_url', node, queryParams);
+      queryParams.set('min_age', minAge ?? '');
+      queryParams.set('max_age', maxAge ?? '');
+      setQueryParam('sex', sex, queryParams);
+      setQueryParam('diagnosis', isControl ? null : diagnosis, queryParams);
+      queryParams.set('is_control', isControl ? 'true' : '');
+      queryParams.set('min_num_sessions', minNumSessions ?? '');
+      setQueryParam('assessment', assessmentTool, queryParams);
+      setQueryParam('image_modal', imagingModality, queryParams);
+  
+      // Notes:
+      // 1. Deleting elements in an array as we loop over it is not good, either make a new object or filter (same thing)
+      // 2. using forEach on the QueryParams object, 
+      // 3. Do the filtering first / switch before adding 
+      // Solution:
+      // Push the keys to be deleted inside keysToDelete and loop over them and delete them from queryParams afterwards
+      const keysToDelete: string[] = [];
+  
+      queryParams.forEach((value, key) => {
+        // if All option is selected for nodes field, delete all node_urls
+        if (value === '' || value === 'allNodes') {
+          keysToDelete.push(key);
+        }
+      });
+  
+      keysToDelete.forEach((key) => {
+        queryParams.delete(key);
+      });
+  
+      return `${apiQueryURL}${queryParams.toString()}`;
+    }
+  
+    return (
+      <div className={isFederationAPI ? 'grid grid-cols-2 grid-rows-8 gap-2' : 'grid grid-cols-2 grid-rows-7 gap-2'}>
+        {
+        isFederationAPI && (
+        <div className='col-span-2'>
+          <CategoricalField 
+          label='Neurobagel graph'
+          options={nodeOptions.map((n) => ({
+            label: n.NodeName,
+            id: n.ApiURL
+          }))}
+          onFieldChange={(label, value) => updateCategoricalQueryParams(label, value)} 
+          multiple
+          inputValue={node}
```

Comment:
hmmm I'm not sure about that because the naming is in a way standardized across different fields and this would be the odd one out.
I did rename a couple of things in one of the previous PRs during refactoring, do you think that may have caused the confusion?